### PR TITLE
added field export_url in expense group and util to generate URL

### PR DESCRIPTION
**Pull request by Ashutosh619-sudo**

### CODE REVIEW

File: scripts/python/update-export-url.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+expense_groups = ExpenseGroup.objects.all()
+
+for expense_group in expense_groups:
+    try:
+        netsuite_cred = NetSuiteCredentials.objects.get(workspace_id=expense_group.workspace_id)
+        url = generate_netsuite_export_url(response_logs=expense_group.response_logs, ns_account_id=netsuite_cred.ns_account_id)
+        expense_group.export_url = url
+        expense_group.save()
+        print('Export URl updated for expense group id {}'.format(expense_group.id))
```
Comment (NileshPant1999) [inline scripts/python/update-export-url.py:14]: remove loggers

File: scripts/python/update-export-url.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    except Exception as exception:
+        print('Something went wrong during updating export_url for workspace_id {}'.format(expense_group.workspace_id))
```
Comment (NileshPant1999) [inline scripts/python/update-export-url.py:16]: add valid exceptions here

File: scripts/python/update-export-url.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from apps.fyle.models import ExpenseGroup
+from apps.workspaces.models import NetSuiteCredentials
+from fyle_netsuite_api.utils import generate_netsuite_export_url
+
+
+expense_groups = ExpenseGroup.objects.all()
+
+for expense_group in expense_groups:
```
Comment (ashwin1111) [inline scripts/python/update-export-url.py:8]: let's iterate through workspaces (prod only - exclude workspaces that has name like test / fyle / sandbox). Iterate through workspace and filter expense groups that has response logs != null and do it in batches. https://github.com/fylein/fyle-integration-scripts/blob/master/qbo/batched-fill.py check this out on how to do it in batches

File: fyle_netsuite_api/utils.py [hunk 4]
```diff
@@ -23,3 +34,16 @@ def filter_queryset(self, queryset):
...
+def generate_netsuite_export_url(response_logs, ns_account_id):
+    if response_logs:
+        try:
+            export_type = response_logs['type'] if response_logs['type'] else 'chargeCard'
+            internal_id = response_logs['internalId']
+            redirection = EXPORT_TYPE_REDIRECTION[export_type]
+            url = f'https://{ns_account_id}.app.netsuite.com/app/accounting/transactions/${redirection}.nl?id={internal_id}'
+            return url
+        except Exception as exception:
+            logger.info({'error': exception})
```
Comment (ashwin1111) [inline fyle_netsuite_api/utils.py:48]: logger.exception

File: scripts/python/update-export-url.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+from apps.fyle.models import ExpenseGroup
+from apps.workspaces.models import NetSuiteCredentials, Workspace
+from fyle_netsuite_api.utils import generate_netsuite_export_url
+
+
+folder_created_workspace_ids = []
```
Comment (ashwin1111) [inline scripts/python/update-export-url.py:6]: not needed

File: scripts/python/update-export-url.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+prod_workspaces = Workspace.objects.exclude(
+    name__iregex=r'(fyle|test)',
+    id__in=folder_created_workspace_ids
```
Comment (ashwin1111) [inline scripts/python/update-export-url.py:10]: not needed

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**